packages/circuit-ui/components/ComparisonTable/components/PlanTable/PlanTable.tsx

```diff
@@ -108,28 +107,32 @@ export const PlanTable = forwardRef<HTMLTableElement, PlanTableProps>(
         0,
       ) > COLLAPSE_THRESHOLD,
     );
-    const [headerHeight, setHeaderHeight] = useState(0);
+    const [sectionOffset, setSectionOffset] = useState(0);
     const isPlanPickerVisible = headers.length > 2;
-    const offset =
-      headerHeight +
-      (isPlanPickerVisible ? (isMobile ? 80 : 0) + (isTablet ? 16 : 0) : 0);
-
-    const updateHeaderHeight = useCallback(() => {
-      throttle(() => {
-        setHeaderHeight(theadRef.current?.getBoundingClientRect().height ?? 0);
-      }, 500)();
-    }, []);
 
     useEffect(() => {
-      updateHeaderHeight();
-    }, [updateHeaderHeight]);
+      if (!theadRef.current || !ResizeObserver) {
+        return undefined;
+      }
 
-    useEffect(() => {
-      window.addEventListener('resize', updateHeaderHeight);
+      // opt for progressive enhancement
+      // eslint-disable-next-line compat/compat
+      const headerSizeObserver = new ResizeObserver((entries) => {
+        setSectionOffset(
+          entries[0].contentRect.height +
+            (isPlanPickerVisible
+              ? (isMobile ? 80 : 0) + (isTablet ? 16 : 0)
+              : 0),
+        );
+      });
+
+      headerSizeObserver.observe(theadRef.current);
       return () => {
-        window.removeEventListener('resize', updateHeaderHeight);
+        if (theadRef.current) {
+          headerSizeObserver.unobserve(theadRef.current);
+        }
```

Comment:
Accessing `theadRef.current` this way can cause errors or a memory leak since `theadRef.current` could change between the execution of the effect and its cleanup function. 

To avoid this, assign `theadRef.current` to a local variable at the beginning of the effect and use it in the cleanup function.